htmresearch/frameworks/nlp/classify_htm.py

```diff
@@ -104,6 +89,32 @@ def trainToken(self, token, labels, sampleId, reset=0):
       self.reset()
 
 
+  def trainDocument(self, document, labels, sampleId):
+    """
+    Train the model with the given document, associated labels, and sampleId.
+    This routine will tokenize the document and train the model on these tokens
+    using the given labels and ID. A reset will be issued after the document has
+    been trained.
+
+    @param document   (str)  The text token to train on.
+    @param labels     (list) A list of one or more integer labels associated
+                             with this token.
+    @param sampleId   (int)  An integer ID associated with the entire document.
+    """
+    # Train on each token with an id that points to its corresponding word in
+    # the original text. This id is generated by multiplying the document's id
+    # by an indexing factor (e.g. 1000), and then adding the mapping index (from
```

Comment:
Yes that's better, but then the client would become a little more complicated by calling `trainToken()` for some models and `trainDocument()` for others.